### DISCUSSION TITLE
When next release is planned?

### DISCUSSION THREAD

**Discussion by liranHaim** [libssh2/libssh2#1051] [Q&A] [answered]
Hello
I have an issue connecting to  OpenSSH_8.7p1 and OpenSSL 3.0.1 servers with the  last version 1.10
I  know that it was fixed in the main branch 
So I wait for the new release with the fix.
Thanks in advance

**Comment by ionutbaltariu** [comment]
I was going to ask the same thing. On the official site it states "The latest release: [libssh2 1.10.0](https://www.libssh2.org/download/libssh2-1.10.0.tar.gz) from August 29th 2021"

That version is kind of old. When can we expect a new release?

**Comment by Zenju** [comment]
For what it's worth, my app is currenly using the master version from 2022-11-11, with no reported issues (*)

(*) except for the pull requests opened by me.

**Comment by sbz** [comment]
Hi,

I would be interested also to know when the next release will be. It's been few years since last [release 1.10](https://github.com/libssh2/libssh2/releases/tag/libssh2-1.10.0) and https://github.com/libssh2/libssh2/issues/572

@dfandrich @vszakats @willco007  @bagder could you help here with a new release?

**Comment by adamyg** [comment]
Interested in formal release for mbedTLS 3.x.x integration.

**Comment by fxcoudert** [comment]
As a maintainer for Homebrew, we would like to ship a more recent release without carrying patches. I know other projects also do, like Julia, which ships patches for https://github.com/libssh2/libssh2/pull/711 and https://github.com/libssh2/libssh2/pull/650

**Comment by dfandrich** [comment]
Under the mantra of "release early, release often" I think a release now makes sense. Yes there are a lot of PRs and issues outstanding, but there are also enough improvements already submitted to warrant a release now. We can always do another release in a few months once some of the remaining known issues have been dealt with.

@willco007 is there anything that *has* to go in now? Current git seems pretty stable. @bagder has volunteered to handle the mechanics of a release once you give the all clear.

**Comment by willco007** [comment]
It's in good shape, nothing else has to go in to this release. Nice to haves may be this outstanding PR https://github.com/libssh2/libssh2/pull/712 and a null check to resolve this: https://github.com/libssh2/libssh2/issues/802 .

**Comment by dfandrich** [comment]
Is there someone around who can approve PR #712? My cmake-fu is pretty slim.

**Comment by jakob** [comment]
@willco007 Since PR #712 was closed and reopened as PR #816, which was merged, it should be good to go now?

I'd love to go back to using official releases, I've been using patched versions for too long.

**Comment by willco007** [comment]
@jakob, @vszakats Is currently cranking away at quality of life improvements long overdue; when that settles down a bit we can get a release together.

**Comment by vszakats** [comment]
These two PRs would be a excellent additions for the new release:
- https://github.com/libssh2/libssh2/pull/655
  https://github.com/libssh2/libssh2/pull/987
- https://github.com/libssh2/libssh2/pull/797

They would allow disabling obsolete/less-secure server features, that are now necessary for clients using libssh2.

**Comment by dfandrich** [comment]
I've just closed #802 as well. #655 and #797 would be great to have in the next release, but I'd rather get a release now without them than a release in a few months (or years, if #655 is any indication) with them. They're both pretty large changes that should be well reviewed first.

**Comment by Un1q32** [comment]
A new release would be useful for me as well.  I have to use a patch to support libressl and it would be nice to drop it.

**Comment by sbz** [comment]
I'm the maintainer On [FreeBSD](https://www.freebsd.org), we also include the [following patch](/freebsd/freebsd-ports/blob/main/security/libssh2/files/patch-src_openssl.h) for libressl 3-5 fix from commit [b952674f](https://github.com/libssh2/libssh2/commit/b952674f). I will be glad to get rid of it with a new release. 

Also, let me know, how I can assist to help validate the new release 🤝

**Comment by vszakats** [comment]
I merged PR #846 dealing with compiler warnings. It brings them down to zero on CI and other tested platforms. 

Functional tests would be welcome with current master, to see if there is any regression. Reports of any outstanding compiler warnings are welcome as well.

**Comment by pm-cfs** [comment]
Wanted to follow up on this. We also really need an official release.
At lot of very useful updates have been completed since Aug 2021, and it would be very helpful to have those available for production systems.

**Comment by doorsdown** [comment]
Unless #655 and #797 are going to be merged by the end of the month I think it would be great if a new version was pushed out in the mean time. Otherwise I can see this getting stuck in a cycle of going of oh it would be great if we included this and then that. Looking back in this ticket we can already see the initial goal posts were #712 and #802 which are closed/merged now. There is a cost in peoples time to make releases so its understandable that they are not ultra frequent after a few features are completed but this seems to be at the point of piling up features preventing them from getting out.

**Comment by vszakats** [comment]
We have an intermittent issue with WinCNG hostkey verification:
#804 and #979. Help with this one would be much appreciated.
Almost every CI run fails because of it.

#655 is on the way too, now rebased onto #797 and tracked under #987.

Testing with recent master and reviews of pending PRs are welcome too.

**Comment by etychon** [comment]
Hello @bagder - with latest changes by @vszakats (many thanks!) it would be really nice to have a new release.
We use libssh2 with Guacamole and would really like to use released versions instead of latest master.

**Comment by bagder** [comment]
I'm happy to do a release, but @vszakats and @willco007 are the ones leading this now primarily so I'm waiting for them to do what they think is necessary and let me know when I can press the buttons to make the release happen.

**Comment by vszakats** [comment]
I've nothing more major pending. Some external testing/reviewing would definitely be useful. We still have 1 open PR #791 that seems useful for inclusion, but needs review.

Upon the release, we might consider announcing the deprecation of these crypto elements, whereas upon a next version they would become disabled by default: `LIBSSH2_MD5`, `LIBSSH2_3DES`, `LIBSSH2_BLOWFISH`, `LIBSSH2_RC4`, `LIBSSH2_DSA`, `LIBSSH2_HMAC_RIPEMD`, `LIBSSH2_CAST` (for now they can disabled manually via their newly added `NO` options). Possibly even anything with `SHA1`. These have been obsolete and considered insecure for many years now.

Also sorry for the long burst of commits.

**Comment by migueldeicaza** [comment]
The PR (#791) would indeed be very useful to have.

Would love to keep the old ciphers around, for interop purposes.   In the end, the challenge is that servers are the ones with that need to be upgraded, and when that happens, users will naturally flow to the more secure connections.  Deprecating might just prevent users from getting to their data in the meantime.

**Comment by MonkeybreadSoftware** [comment]
Yes, please. If you could make a beta release, I'd love to compile it on various targets and try it.
And #791 looks okay to me, so I'd love to try it.

**Comment by Un1q32** [comment]
I tested the git version on iOS 6.1.6 on an iPod touch 4 and I can curl files from ssh servers just fine

**Comment by vszakats** [comment]
Adding more CI tests turned up a regression in wolfSSL builds: #1020

**Comment by Un1q32** [comment]
> I tested the git version on iOS 6.1.6 on an iPod touch 4 and I can curl files from ssh servers just fine

this was with curl 8.0.1 and libressl 3.7.2 by the way

**Comment by pm-cfs** [comment]
Wanted to follow up on this as it feels like we have a moving target. This question was opened 4 mths ago; and was preceded by [#657](https://github.com/libssh2/libssh2/issues/657) back in Jan 2022. It seems like a release is always imminent, but then there is another PR we're waiting for. Some of the bug fixes and feature updates that have been in master for a very long time now are worthy of a dot release on their own; at this point in 2023, there are quite a few of them. Hoping we can get 1.11 in the next month (Jun 2023).
We would very much love to be able to use all the hard work that has already been committed!
Thanks!

**Comment by bagder** [comment]
@vszakats @willco007: how about we set a date and make it happen? Push the changes with updated RELEASE-NOTES and I can push the buttons to: build the tarball, sign the thing, upload to the website, and update the website accordingly. Like perhaps next week?

**Comment by willco007** [comment]
@bagder @vszakats Sounds good, I will build up the release notes this week. How about 5/30 for release?

**Comment by migueldeicaza** [comment]
Music to my ears

**Comment by dfandrich** [comment]
I haven't been able to devote much time to looking into #1020 but it may be significant for WolfSSL users. I'll try to look into it in the next few days.

**Comment by bagder** [comment]
Let's aim for releasing 1.11.0 on May 30. I think that maybe it is worth shipping the release even if we don't manage to fix #1020 before that. The we do a follow-up patch release later with that (and other) fixes.

**Reply by pm-cfs** [reply]
Thank you everyone for 1.11.0 !!!

**Comment by triatic** [comment]
Long overdue. I came here as a result of trying to work out why php's ssh2_auth_pubkey_file() was failing only to find out that libssh2 still does not support RSA SHA256 in release (resolved in master by #536). RSA is the format created by Puttygen by default and modern servers disallow RSA SHA1 by default, so it is most likely a widespread compatibility issue now.

**Comment by vszakats** [comment]
@bagder If you agree, it'd be nice to ship the source package in .zip, .tar.xz and .tar.bz2 formats too upon the next release. I copied the necessary `maketgz` bits from curl for this in c7d48443867e032edbf0f29e576a46fc1dc03804 (#874).

**Comment by bagder** [comment]
Sure, I don't mind that!

**Comment by bagder** [comment, marked as answer]
1.11.0 shipped

**Reply by tmzhuang** [reply]
Thank you guys for all your hard work :)

**Comment by MonkeybreadSoftware** [comment]
Thank you very much!
We'll try it with CURL 8.1.2

**Comment by Un1q32** [comment]
![Screenshot from 2023-05-30 17-50-47](https://github.com/libssh2/libssh2/assets/57970114/4f972ea7-16ab-469b-9b67-5b7fada9d775)

works on my machine

**Comment by etychon** [comment]
Thank you very much for this new and highly anticipated release!

**Comment by sbz** [comment]
Thanks for the release! I pushed the update in our [FreeBSD]() ports tree downstream with this new 1.11.0 release [b9aa342e83](https://github.com/freebsd/freebsd-ports/commit/b9aa342e83). It should be available soon

```
sbz@freebsd:~/git/ports/security/libssh2 $ pkg info|grep libssh2
libssh2-1.11.0,3               Library implementing the SSH2 protocol
```
